src/Integration.UnitTests/Binding/CSharpVBBindingOperationTests.cs

```diff
@@ -416,13 +418,38 @@ public void ProjectBindingOperation_Commit_NewProjectSystem_DoesNotAddFile()
             this.projectMock.Files.ContainsKey(expectedFile).Should().BeFalse("Should not add the file to the project for the new project system");
         }
 
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void ProjectBindingOperation_Commit_ShouldNotGenerateProjectRuleSetWhenRuleSetIsDefault(bool isLegacySystem)
+        {
+            // Arrange
+            var testSubject = CreateTestSubject();
+            projectMock.SetCSProjectKind();
+            var prop = CreateProperty(this.projectMock, "config1", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            testSubject.Initialize();
+            testSubject.Prepare(CancellationToken.None);
+
+            projectSystemHelper.SetIsLegacyProjectSystem(isLegacySystem);
+
+            // Act
+            using (new AssertIgnoreScope()) // Ignore that the file is not on disk
+            {
+                testSubject.Commit();
+            }
+
+            // Assert
+            prop.Value.ToString().Should().Be(PathHelper.CalculateRelativePath(testSubject.ProjectFullPath, cSharpVBBindingConfig.RuleSet.Path), "Should update the property value");
+            projectMock.Files.ContainsKey(cSharpVBBindingConfig.RuleSet.Path).Should().Be(isLegacySystem, "Should add the file to the project only if legacy system");
+        }
+
         [TestMethod]
         public void ProjectBindingOperation_Commit_LegacyProjectSystem_DoesAddFile()
         {
             // Arrange
             CSharpVBBindingOperation testSubject = this.CreateTestSubject();
             this.projectMock.SetCSProjectKind();
-            PropertyMock prop = CreateProperty(this.projectMock, "config1", CSharpVBBindingOperation.DefaultProjectRuleSet);
+            PropertyMock prop = CreateProperty(this.projectMock, "config1", "ConcurrencyRules.ruleset");
```

Comment:
A minor point, but a completely arbitrary name like `ConcurrencyRules.ruleset` doesn't give any clue about why the property is being set in the test.
A name like `MyCustomProjectLevelRules.ruleset` or `NonDefaultRules.ruleset` gives more of a hint as to what the logical purpose of the property is.